File: kt/src/main/kotlin/com/github/scglwsj/parking/boy/OderParkingBoyService.kt
package com.github.scglwsj.parking.boy

import com.github.scglwsj.parking.lot.ParkingLot

class OderParkingBoyService : ParkingBoyService {
    override fun findOneValidParkingLot(parkingLots: List<ParkingLot>): ParkingLot? =
        parkingLots.firstOrNull { it.hasAvailableSpaces }
}

File: cs/Parking/Boy/OderParkingBoyService.cs
using System.Collections.Generic;
using System.Linq;

namespace Parking
{
    class OderParkingBoyService : IParkingBoyService
    {
        public ParkingLot FindOneValidParkingLot(IReadOnlyList<ParkingLot> parkingLots) => 
            parkingLots.FirstOrDefault(parkingLot => parkingLot.HasAvailableSpaces);
    }
}
